### PR TITLE
[Trainer] Report both steps and num samples per second

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

As seen with @stas00, there is a bug in the current speed metrics reporting: training reports the number of training steps per second while evaluation and predict report the number of samples per second. After discussion we concluded that both are interesting, so this PR updates the Trainer to report both.

### CODE REVIEW

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1130,14 +1136,6 @@ def train(
         # self.model_wrapped is DDP(Transformers Model), Deepspeed(Transformers Model), etc.
 
         # Train!
-        if is_torch_tpu_available():
-            world_size = xm.xrt_world_size()
-        elif args.local_rank != -1:
-            world_size = dist.get_world_size()
-        else:
-            world_size = 1
```
Comment (sgugger) [inline src/transformers/trainer.py:1138]: This is all done by `args.world_size`, which is why this part is removed.

File: src/transformers/trainer_utils.py [hunk 2]
```diff
@@ -250,6 +250,9 @@ def speed_metrics(split, start_time, num_samples=None):
     if num_samples is not None:
         samples_per_second = 1 / (runtime / num_samples)
         result[f"{split}_samples_per_second"] = round(samples_per_second, 3)
+    if num_steps is not None:
+        steps_per_second = 1 / (runtime / num_steps)
```
Comment (stas00) [inline src/transformers/trainer_utils.py:254]:
```suggestion
        steps_per_second = num_steps / runtime
```

Won't this be easier to read ;)

same for `num_samples` above?

**Review by stas00 — APPROVED** [review]
Awesome improvement, @sgugger - thank you!

File: src/transformers/trainer_utils.py (hunk 2 above)
Comment (sgugger) [reply src/transformers/trainer_utils.py:254]: I just copy-pasted without thinking ;-) This is better indeed, will do the other one.

**Pull request merged**